### DISCUSSION TITLE
[general] Error with crew package

### DISCUSSION THREAD

**Discussion by robschick** [ropensci/targets#1101] [General]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

When I use the `crew` package with my targets workflow, e.g.

```
library(crew)
tar_option_set(
  controller = crew::crew_controller_local(workers = 2)
)
```

I get this error:

```
Error:
! Error running targets::tar_make()
  Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
  Debugging guide: https://books.ropensci.org/targets/debugging.html
  How to ask for help: https://books.ropensci.org/targets/help.html
  Last error: invalid daemons: structure(5L, class = "errorValue")
```

To recreate this, I invalidated a target that ran, and then re-issued tar_make()

```
tar_invalidate(pts_mult_imp)
tar_make(pts_mult_imp)
```

If I don't use the above syntax for `crew`, the same target runs fine:

```
▶ start target pts_mult_imp
● built target pts_mult_imp [42.639 seconds]
▶ end pipeline [54.853 seconds]
```

@wlandau - The repo is private right now ahead of two manuscripts that it serves, but am happy to send what I can from say the _targets.R file.

thanks in advance,
Rob

**Comment by wlandau** [comment]
Thanks for reporting this. @shikokuchuo and I have spent a lot of time trying to reproduce and isolate it, and it's tough to do from our end.

How often/reliably does the daemons error 5 happen? How small/simple can you make your `targets` pipeline and still have it reproduce the error? Would you be able to post a complete reprex and information about your platform and system, e.g. session info? Please see https://books.ropensci.org/targets/help.html#write-a-reprex

**Reply by robschick** [reply]
Yes, I will be happy to try this. I've only seen it creep in when I made that change to the full workflow. I'll try to make a reprex, but will take me a while. Since this runs without it, I need to get the papers out the door first ;-) 

thanks for the quick reply

**Reply by wlandau** [reply]
Thanks. Yes, this is definitely a crew and/or mirai issue and not a targets issue, but still important to track down.

**Reply by shikokuchuo** [reply]
This is https://github.com/wlandau/crew/issues/76 appearing again it seems. I didn't realise this was still an issue. I am going to upgrade the dispatcher monitor sockets in `mirai`. I daresay that that will eliminate these errors once and for all.

**Reply by shikokuchuo** [reply]
@robschick I hope to save you the trouble of producing a reprex if, by installing the latest development version of `mirai` from R-universe, your pipeline is able to run?
```r
install.packages("mirai", repos = "shikokuchuo.r-universe.dev")
```
Apologies in case it doesn't help, in which case a reprex would be greatly appreciated as we are currently lacking a concrete example, and hence can only proceed theoretically.

**Reply by robschick** [reply]
Thanks! Got this error back, what is needed for install? I have Xcode installed

```
No existing 'libmbedtls' >= 2 found
Detecting 'cmake'...
Required 'cmake' not found
```

**Reply by robschick** [reply]
success after installing cmake!

```
✔ skip pattern modelfits
▶ start target pts_mult_imp
● built target pts_mult_imp [44.385 seconds]
▶ end pipeline [1.134 minutes]
```

thanks both for your help

**Reply by wlandau** [reply]
Awesome! Success installing mirai, or the original issue, or both?

Thanks @shikokuchuo for the patch!

**Reply by robschick** [reply]
Both! thanks again
